### PR TITLE
Upgrade to Keycloak 14.0.0

### PR DESCRIPTION
**Pull request by fperot74**

### CODE REVIEW

File: cloudtrust-common/src/test/java/io/cloudtrust/utils/CryptoUtilTest.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public void init() {
     public void testSecretEncryptionDecryption() {
         byte[] plainText = new byte[32]; //256 bits
         new SecureRandom().nextBytes(plainText);
-        assertThat(CryptoUtil.decryptFromDatabaseStorage(CryptoUtil.encryptForDatabaseStorage(plainText)), equalTo(plainText));
+        org.hamcrest.MatcherAssert.assertThat(CryptoUtil.decryptFromDatabaseStorage(CryptoUtil.encryptForDatabaseStorage(plainText)), equalTo(plainText));
```
Comment (lagess) [inline cloudtrust-common/src/test/java/io/cloudtrust/utils/CryptoUtilTest.java:40]: Why have the full class package, as it is anyway imported?

File: kc-cloudtrust-module/kc-cloudtrust-common/src/main/java/io/cloudtrust/keycloak/FormUtils.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import org.apache.commons.lang3.tuple.Pair;
+import org.keycloak.authentication.AuthenticationFlowContext;
+import org.keycloak.authentication.AuthenticationSelectionOption;
+import org.keycloak.credential.CredentialProvider;
+import org.keycloak.credential.CredentialTypeMetadata;
+import org.keycloak.credential.CredentialTypeMetadataContext;
+import org.keycloak.forms.login.LoginFormsProvider;
+import org.keycloak.models.AuthenticationExecutionModel;
+
+public class FormUtils {
```
Comment (lagess) [inline kc-cloudtrust-module/kc-cloudtrust-common/src/main/java/io/cloudtrust/keycloak/FormUtils.java:18]: I'd love some doc about the goal of this class...

**Review by lagess — CHANGES_REQUESTED** [review]

File: cloudtrust-common/src/test/java/io/cloudtrust/utils/CryptoUtilTest.java (hunk 1 above)
Comment (fperot74) [reply cloudtrust-common/src/test/java/io/cloudtrust/utils/CryptoUtilTest.java:40]: Absolutely no idea!

File: kc-cloudtrust-module/kc-cloudtrust-common/src/main/java/io/cloudtrust/keycloak/FormUtils.java (hunk 2 above)
Comment (fperot74) [reply kc-cloudtrust-module/kc-cloudtrust-common/src/main/java/io/cloudtrust/keycloak/FormUtils.java:27]: Class commented

Comment (lagess) [reply kc-cloudtrust-module/kc-cloudtrust-common/src/main/java/io/cloudtrust/keycloak/FormUtils.java:27]: Thanks :-)

**Review by lagess — APPROVED** [review]
Seems fine for me

**Review by harture — APPROVED** [review]

**Pull request merged**